TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/WorkspacesController.php

```diff
@@ -395,7 +395,7 @@ public function publishWorkspaceAction(Workspace $workspace)
         if (($targetWorkspace = $workspace->getBaseWorkspace()) === null) {
             $targetWorkspace = $this->workspaceRepository->findOneByName('live');
         }
-        $workspace->publish($targetWorkspace);
+        $this->publishingService->publishNodes($this->publishingService->getUnpublishedNodes($workspace), $targetWorkspace);
```

Comment:
Wouldn't it be better to use `Workspace.publish()` instead also in the `WorkspaceController`? Publishing a workspace should be something you do on the workspace, and if needed _that_ can use a service to actually do the task.

Or is there a reason other than "make it like in the `WorkspaceController` for this change?